src/Orleans.Streaming/Checkpointers/StreamCheckpointerGrain.cs

```diff
@@ -0,0 +1,29 @@
+using Orleans.Placement;
+using Orleans.Providers;
+using Orleans.Runtime;
+using System.Threading.Tasks;
+
+namespace Orleans.Streams
+{
+    [PreferLocalPlacement]
+    public class StreamCheckpointerGrainGrain : Grain, IStreamCheckpointerGrain
+    {
+        private readonly IPersistentState<StreamCheckpointerGrainState> _state;
```

Comment:
I'm here to drop a very trivial question : )
Is there any reason not to use the native checkpoint of Kinesis?

Reply:
@scalalang2 to my knowledge, Kinesis doesn't have the ability to "keep track" of the current position for a consumer so that it can resume from a specified position if the consumer is stopped.
Kafka has the capability (I think it's called "Commit").

But if you can point me in the direction of the Kinesis API you have in mind, I'm happy to take a look.

Reply:
@dpbevin 
Thanks for reply : )

> to my knowledge, Kinesis doesn't have the ability to "keep track" of the current position for a consumer

You're right, I misunderstood Kinesis natively has concept of "checkpoint".

It uses DynamoDB as lease table to save checkpoint [sample code](https://github.com/awslabs/amazon-kinesis-client-net/blob/master/SampleConsumer/SampleConsumer.cs#L135-L144) [official docs](https://docs.aws.amazon.com/streams/latest/dev/shared-throughput-kcl-consumers.html),

Reply:
Appreciate the info.

I _could_ have added a DynamoDB checkpointer, which would have been similar to how the Azure EventHub streaming uses Azure Table storage for checkpoints but I felt that a grain-based solution would work in more places and... my intended use of Kinesis streaming won't be able to make use of DynamoDB.

Reply:
I also agree this, thx : )
> I felt that a grain-based solution would work in more places